code/modules/paperwork/photography.dm

```diff
@@ -175,119 +175,141 @@
+			CHECK_TICK
 	return res
 
+/obj/item/device/camera/proc/get_mob_descriptions(turf/the_turf, existing_descripion)
+	var/mob_detail = existing_descripion
+	for(var/mob/living/carbon/cur_carbon in the_turf)
+		if(cur_carbon.invisibility)
+			continue
 
-/obj/item/device/camera/proc/get_mobs(turf/the_turf as turf)
-	var/mob_detail
-	for(var/mob/living/carbon/A in the_turf)
-		if(A.invisibility) continue
 		var/holding = null
-		if(A.l_hand || A.r_hand)
-			if(A.l_hand) holding = "They are holding \a [A.l_hand]"
-			if(A.r_hand)
+		if(cur_carbon.l_hand || cur_carbon.r_hand)
+			if(cur_carbon.l_hand)
+				holding = "They are holding \a [cur_carbon.l_hand]"
+			if(cur_carbon.r_hand)
 				if(holding)
-					holding += " and \a [A.r_hand]"
+					holding += " and \a [cur_carbon.r_hand]"
 				else
-					holding = "They are holding \a [A.r_hand]"
+					holding = "They are holding \a [cur_carbon.r_hand]"
+
+		var/hurt = ""
+		if(cur_carbon.health < 75)
+			hurt = prob(25) ? " - they look hurt" : " - [cur_carbon] looks hurt"
 
 		if(!mob_detail)
-			mob_detail = "You can see [A] on the photo[A:health < 75 ? " - [A] looks hurt":""].[holding ? " [holding]":"."]. "
+			mob_detail = "You can see [cur_carbon] in the photo[hurt].[holding ? " [holding]" : "."]."
 		else
-			mob_detail += "You can also see [A] on the photo[A:health < 75 ? " - [A] looks hurt":""].[holding ? " [holding]":"."]."
+			mob_detail += " You [prob(50) ? "can" : "also"] see [cur_carbon] in the photo[hurt].[holding ? " [holding]" : "."]."
 	return mob_detail
 
 /obj/item/device/camera/afterattack(atom/target as mob|obj|turf|area, mob/user as mob, flag)
-	if(!on || !pictures_left || ismob(target.loc) || isstorage(target.loc)) return
-	if(user.contains(target) || istype(target, /atom/movable/screen)) return
-	captureimage(target, user, flag)
+	if(!on || !pictures_left || ismob(target.loc) || isstorage(target.loc))
+		return
+	if(user.contains(target) || istype(target, /atom/movable/screen))
+		return
 
 	playsound(loc, pick('sound/items/polaroid1.ogg', 'sound/items/polaroid2.ogg'), 15, 1)
 
 	pictures_left--
 	desc = "A polaroid camera. It has [pictures_left] photos left."
 	to_chat(user, SPAN_NOTICE("[pictures_left] photos left."))
+
+	captureimage(target, user, flag)
+
 	icon_state = icon_off
 	on = 0
 	spawn(64)
 		icon_state = icon_on
 		on = 1
 
 /obj/item/device/camera/proc/captureimage(atom/target, mob/user, flag)
-	var/mobs = ""
+	var/mob_descriptions = ""
 	var/radius = (size-1)*0.5
 	var/list/turf/turfs = RANGE_TURFS(radius, target) & view(world_view_size + radius, user.client)
-	for(var/turf/T as anything in turfs)
-		mobs += get_mobs(T)
-	var/datum/picture/P = createpicture(target, user, turfs, mobs, flag)
-	printpicture(user, P)
+	for(var/turf/the_turf as anything in turfs)
+		mob_descriptions = get_mob_descriptions(the_turf, mob_descriptions)
+	var/datum/picture/the_picture = createpicture(target, user, turfs, mob_descriptions, flag)
+	printpicture(user, the_picture)
```

Comment:
`user` might have been deleted during `createpicture` in which case you want to discard the whole process rather than printing a picture and putting it into the nullspaced actor